packages/features/users/components/UserTable/UserListTable.tsx

```diff
@@ -0,0 +1,353 @@
+  const totalDBRowCount = data?.pages?.[0]?.meta?.totalRowCount ?? 0;
+  const totalFetched = flatData.length;
+
+  //called on scroll and possibly on mount to fetch more data as the user scrolls and reaches bottom of table
+  const fetchMoreOnBottomReached = useCallback(
+    (containerRefElement?: HTMLDivElement | null) => {
+      if (containerRefElement) {
+        const { scrollHeight, scrollTop, clientHeight } = containerRefElement;
+        //once the user has scrolled within 300px of the bottom of the table, fetch more data if there is any
+        if (scrollHeight - scrollTop - clientHeight < 300 && !isFetching && totalFetched < totalDBRowCount) {
+          fetchNextPage();
+        }
+      }
+    },
+    [fetchNextPage, isFetching, totalFetched, totalDBRowCount]
+  );
+
+  useEffect(() => {
+    fetchMoreOnBottomReached(tableContainerRef.current);
+  }, [fetchMoreOnBottomReached]);
+
+  const removeMemberMutation = trpc.viewer.teams.removeMember.useMutation({
+    async onSuccess() {
+      await utils.viewer.teams.get.invalidate();
+      await utils.viewer.eventTypes.invalidate();
+      await utils.viewer.organizations.listMembers.invalidate();
+      showToast(t("success"), "success");
+    },
+    async onError(err) {
+      showToast(err.message, "error");
+    },
+  });
+
+  return (
+    <>
+      <DataTable
+        searchKey="member"
+        selectionOptions={[
+          {
+            label: "Add To Team",
+            onClick: () => {
+              console.log("Add To Team");
+            },
+            icon: Users,
+          },
+          {
+            label: "Delete",
+            onClick: () => {
+              console.log("Delete");
+            },
+            icon: StopCircle,
+          },
+        ]}
+        tableContainerRef={tableContainerRef}
+        tableCTA={
+          adminOrOwner && (
+            <Button
+              type="button"
+              color="primary"
+              StartIcon={Plus}
+              size="sm"
+              className="rounded-md"
+              onClick={() =>
+                dispatch({
+                  type: "INVITE_MEMBER",
+                  payload: {
+                    showModal: true,
+                  },
+                })
+              }
+              data-testid="new-organization-member-button">
+              {t("add")}
+            </Button>
+          )
+        }
+        columns={memorisedColumns}
+        data={flatData}
+        isLoading={isLoading}
+        onScroll={(e) => fetchMoreOnBottomReached(e.target as HTMLDivElement)}
+        filterableItems={[
+          {
+            tableAccessor: "role",
+            title: "Role",
+            options: [
+              { label: "Owner", value: "OWNER" },
+              { label: "Admin", value: "ADMIN" },
+              { label: "Member", value: "MEMBER" },
+            ],
+          },
+        ]}
```

Comment:
For pre-defined filters we can add these in automatically

(Looking for a way to generate this based on the data pulled from api but will come in future iterations)